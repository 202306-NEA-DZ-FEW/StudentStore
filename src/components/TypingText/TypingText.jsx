const { useEffect, useState } = require("react");

const TypingText = ({ text, speed }) => {
    const [displayText, setDisplayText] = useState("");

    useEffect(() => {
        const interval = setInterval(() => {
            if (displayText.length < text.length) {
                setDisplayText(text.substring(0, displayText.length + 1));
            } else {
                clearInterval(interval);
            }
        }, speed);
        return () => clearInterval(interval);
    }, [displayText, text, speed]);

    return (
        <p className='font-poppins leading-loose font-normal text-2xl text-[#585785] '>
            <strong>{displayText}</strong>
        </p>
    );
};

export default TypingText;
