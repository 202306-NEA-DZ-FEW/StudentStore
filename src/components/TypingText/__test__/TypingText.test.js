import renderer from "react-test-renderer";

import TypingText from "../TypingText";

it("renders correctly", () => {
    const tree = renderer.create(<TypingText />).toJSON();
    expect(tree).toMatchSnapshot();
});
