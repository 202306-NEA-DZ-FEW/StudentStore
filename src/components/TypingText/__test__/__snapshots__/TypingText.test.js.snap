// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<p
  className="font-poppins leading-loose font-normal text-2xl text-[#585785] "
>
  <strong />
</p>
`;
